diff --git a/masquerade.module b/masquerade.module
index ca26743..1cfd547 100644
--- a/masquerade.module
+++ b/masquerade.module
@@ -844,7 +844,7 @@ function masquerade_switch_user($uid) {
   }
 
   // Call logout hooks when switching from original user.
-  module_invoke_all('user_logout', $user);
+  masquerade_user_logout($user);
   drupal_session_regenerate();
 
   $query = db_insert('masquerade');
@@ -904,7 +904,7 @@ function masquerade_switch_back() {
   $oldname = ($user->uid == 0 ? variable_get('anonymous', t('Anonymous')) : $user->name);
 
   // Call logout hooks when switching from masquerading user.
-  module_invoke_all('user_logout', $user);
+  masquerade_user_logout($user);
   drupal_session_regenerate();
 
   $user = user_load($uid);
